examples/inspection/plot_linear_model_coefficient_interpretation.py

```diff
@@ -0,0 +1,274 @@
+# The model learnt is far to be a good model making accurate prediction.
+# As interpretation tools characterize model rather than the generative process
+# of the data itself, it needs to be emphasized that interpretations are
+# correct if the model is correct as well.
+#
+# Interpreting coefficients
+# .........................
+#
+# First of all, we can plot the values of the coefficients of the regressor we
+# have fitted.
+
+feature_names = (model.named_steps['columntransformer']
+                      .named_transformers_['onehotencoder']
+                      .get_feature_names(input_features=categorical_columns))
+feature_names = np.concatenate(
+    [feature_names, binary_columns, numerical_columns])
+
+coefs = pd.DataFrame(
+    model.named_steps['transformedtargetregressor'].regressor_.coef_,
+    columns=['Coefficients'], index=feature_names
+)
+coefs.plot(kind='barh', figsize=(9, 7))
+plt.axvline(x=0, color='.5')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# Soon we realize that we cannot compare different coefficients since we did
+# not scale the data before the fit, features having different value ranges.
+# For instance, the "AGE" coefficient is expressed in $/hours/leaving years
+# while the "EDUCATION" is expressed in $/hours/years of education.
+# This is evident if we compare feature standard deviations.
+
+X_train_preprocessed = pd.DataFrame(
+    model.named_steps['columntransformer'].transform(X_train),
+    columns=feature_names
+)
+X_train_preprocessed.std().plot(kind='barh', figsize=(9, 7))
+plt.title('Features std. dev.')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# We should then normalize the coefficients by the standard deviation and we
+# will be able to compare them.
+
+coefs = pd.DataFrame(
+    model.named_steps['transformedtargetregressor'].regressor_.coef_ *
+    X_train_preprocessed.std(),
+    columns=['Coefficients'], index=feature_names
+)
+coefs.plot(kind='barh', figsize=(9, 7))
+plt.axvline(x=0, color='.5')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# The plot above tells us that an increase of the "AGE" will induce a decrease
+# of the "WAGE" when all other features remain constant. Also an increase of
+# the "EXPERIENCE" will induce an increase of the "WAGE" when all other
+# features remain constant.
+#
+# The first interpretation might look counter-intuitive at first, if one
+# relates the relationship between "AGE" and "WAGE" as a marginal link.
+# However, as previously mentioned, a linear model computes a conditional
+# link between "AGE" and "WAGE" given all other features.
+# Therefore, one should also interpret that for a given experience and all
+# other features constant as well, a younger person would have an higher
+# wage.
+#
```

Comment:
As discussed IRL, we need a new section regarding pre-scaling the data within the pipeline and the effect of regularization on the coefficients obtained. We will different results then and it will be an interesting thing to discuss.